src/main/scala/tilelink/Monitor.scala

```diff
@@ -517,6 +599,98 @@ class TLMonitor(args: TLMonitorArgs, monitorDir: MonitorDirection = MonitorDirec
     when (bundle.a.fire() || bundle.d.fire()) { watchdog := 0.U }
   }
 
+  def legalizeADSourceStrict(bundle: TLBundle, edge: TLEdge) {
```

Comment:
rename to legalizeADSource (since everybody should be using this one)

add depricated to the current legalizeADSource and rename it to something else that shows it's only a workaround for almond